### PR TITLE
[dv,flash_ctrl] regression fix : mp_region test

### PR DESCRIPTION
**Pull request by jdonjdon**
- Add region enable to p2r map update
- Remove naked wait
- Add wait alert checker status complete at the end of the loop.
Signed-off-by: Jaedon Kim <jdonjdon@google.com>

### CODE REVIEW

File: hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_base_vseq.sv
Comment (tjaychen) [inline hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_base_vseq.sv:1092]: ah is this the issue we were discussing yesterday?

**Review by tjaychen — APPROVED** [review]
lgtm. We should merge after getting at least one dv approval.

File: hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_base_vseq.sv
Comment (jdonjdon) [reply hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_base_vseq.sv:1092]: yes

File: hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_mp_regions_vseq.sv
Comment (matutem) [inline hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_mp_regions_vseq.sv:204]: What does "a inside {!(b)}" mean in line 208 below? Perhaps "!a inside {(b)}" is clearer?

File: hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_mp_regions_vseq.sv
Comment (matutem) [inline hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_mp_regions_vseq.sv:215]: So the test was not checking for alerts to complete before this change?

File: hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_mp_regions_vseq.sv
Comment (jdonjdon) [reply hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_mp_regions_vseq.sv:204]: @matutem This is standard negation syntax. Everything but FlashPartData.

File: hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_mp_regions_vseq.sv
Comment (jdonjdon) [reply hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_mp_regions_vseq.sv:215]: @matutem Intention of the test was, send transaction with various region profile.
But original test randomize cfg profile and transaction profile independently s.t. majority of transaction
become illegal.
And set exp_exected_alert at the beginning of the test to allow illegal transaction.
Problem of exp_alert is, 
you can only set the beginning of the test. But test like this, mixed traffic of legal and illegal,
set exp_alert at the beginning provide too coarse pass criteria.
So I updated tb s.t each sequence can update exp_alert dynamically based on its traffic prediction.

File: hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_mp_regions_vseq.sv
Comment (matutem) [reply hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_mp_regions_vseq.sv:204]: Sure, but it is funny to see it within the "set" in an inside operator. In fact, negation is boolean, so perhaps {!2} is just {0}? I think !(a inside {b}) is clearer: see section 18.5.3 in the SV LRM.

Comment (jdonjdon) [reply hw/ip/flash_ctrl/dv/env/seq_lib/flash_ctrl_mp_regions_vseq.sv:204]: @matutem Oh, I made mistake. I thought it was  a !(inside {b}); while I am fixing as you suggested,
my current sb only support single value 'FlashPartInfo'. So I had to change this to
flash_op.partitin == FlashPartInfo. Hope it is ok with you :)

**Comment by jdonjdon** [comment]
> A bit out of scope, but you may want to look into it: the flash_ctrl testplan mentions a region_range covergroup, but I see nothing by that name. Is it missing?

Good catch. this covergroup never exist and I don't think it is necessary. I am updating testplan.

**Pull request merged**
